### ISSUE TITLE
the ‘Experimental features’ is blocked while using Proxy extension in Chrome

### ISSUE THREAD

**Issue by notxx** [uBlock-LLC/uBlock#335]
while using Proxy SwitchySharp, the 'Experimental features' is blocked. is there any way to get these features working?

**Comment by gorhill** [comment]
What do you mean "is blocked"? The net requests to fetch the remote resources are blocked?

**Comment by notxx** [comment]
i think the requests are not handled by uBlock. and when they are handled by the proxy extension, connections reseted, thanks to the GFW.

**Comment by gorhill** [comment]
Ok I checked with Proxy SwitchySharp and a local proxy, and all work fine. Just to be sure I understand the problem:
- The net requests which are supposed to be mirrored are not mirrored?
- These net requests reach their destination fine when not using a proxy?
- These net requests do not reach their destination when using a proxy?

I wasn't aware of Proxy SwitchySharp, so looking at it, it does look quite configurable. Do you have switch rules?

Thing is, uBlock doesn't care or know about how the connection is done, it is just notified when request are made, and in case of mirroring, it will perform its own requests to fetch and cache locally remote resources when these are identified as "mirrorable".

So if I understand the problem, this is where the problem occurs, uBlock's own requests to remote resources are cancelled. So that would be whatever proxy you are using which does that, for whatever reasons.

I suspect the HTTP headers of uBlock's own requests look different than "normal" requests (no referer maybe, etc) and maybe this is what the proxy doesn't like?

**Comment by notxx** [comment]
when i file this issue, i have noticed that a request made to "http://ajax.googleapis.com/ajax/libs/jquery/1.4.2/jquery.min.js" has a 307 response code even i force chrome to refresh (Shift+Refresh). and when Proxy SwitchySharp is enabled, the response is either reseted (with no valid proxy) or established with code 200 or 304 (with valid proxy). that is to say, Proxy SwithySharp is handling the request before uBlock.
is it possible uBlock never get the notification when there is a proxy extension?

**Comment by gorhill** [comment]
> is it possible uBlock never get the notification when there is a proxy extension?

I don't think so. Proxy SwithySharp doesn't handle net requests itself, it just shuffle proxy settings, optionally according to the URL which appears in the address bar (cool feature). There is no intercept of net requests by it. (Even if it was, the chrome API let each extension inspect the net requests in a parallel way, not sequential way).

Do you have any other net-request-related extension installed?

**Comment by notxx** [comment]
after an automatic chrome update, everything turn out to be fine. 307 response with or with-out proxy extension. maybe some internal bug caused my problem?
